src/exception/repo/program.ts

```diff
@@ -16,86 +16,69 @@
+import { DatabaseError } from '../error-handling';
+import { ExceptionValue, ProgramException } from '../types';
 
 const L = loggerFor(__filename);
 
 const programExceptionSchema = new mongoose.Schema<ProgramException>({
   programId: String,
   exceptions: [
     {
+      program_name: String,
       schema: String,
-      coreField: String,
-      exceptionValue: { type: String, enum: Object.values(ExceptionValue) },
+      requested_core_field: String,
+      requested_exception_value: { type: String, enum: Object.values(ExceptionValue) },
     },
   ],
 });
 
-export const ProgramExceptionModel = mongoose.model<ProgramException>(
-  'ProgramException',
-  programExceptionSchema,
-);
+// check if model exists already to account for file watchers eg. test runner with live reload
+const ProgramExceptionModel =
+  mongoose.models.ProgramException ||
+  mongoose.model<ProgramException>('ProgramException', programExceptionSchema);
 
-type RepoResponse = Promise<ProgramException | RepoError>;
-
-export const RepoError = {
-  DOCUMENT_UNDEFINED: 'DOCUMENT_UNDEFINED',
-  SERVER_ERROR: 'SERVER_ERROR',
-} as const;
-
-export type RepoError = ObjectValues<typeof RepoError>;
-
-export interface ProgramExceptionRepository {
-  save(exception: ProgramException): RepoResponse;
-  find(programId: string): RepoResponse;
-  delete(programId: string): RepoResponse;
-}
-
-const checkDoc = (doc: null | ProgramException): RepoError | ProgramException => {
-  if (doc === null || (isArray(doc) && doc.length === 0)) {
-    return RepoError.DOCUMENT_UNDEFINED;
-  }
-  return doc;
-};
-
-export const programExceptionRepository: ProgramExceptionRepository = {
-  async save(exception: ProgramException) {
+const programExceptionRepository = {
+  async save(exception: ProgramException): Promise<ProgramException> {
     L.debug(`Creating new program exception with: ${JSON.stringify(exception)}`);
     try {
-      return await ProgramExceptionModel.findOneAndUpdate(
+      const doc = await ProgramExceptionModel.findOneAndUpdate(
         { programId: exception.programId },
         exception,
         { upsert: true, new: true, overwrite: true },
       ).lean(true);
-      // L.info(`doc created ${doc}`);
+      L.info(`doc created ${JSON.stringify(doc)}`);
+
+      return doc;
     } catch (e) {
       L.error('failed to create program exception: ', e);
-      return RepoError.SERVER_ERROR;
+      throw new DatabaseError('Cannot save program exception.');
     }
   },
 
-  async find(programId: string) {
+  async find(programId: string): Promise<ProgramException | null> {
     L.debug(`finding program exception with id: ${JSON.stringify(programId)}`);
     try {
       const doc = await ProgramExceptionModel.findOne({ programId }).lean(true);
-      return checkDoc(doc);
+      return doc;
     } catch (e) {
       L.error('failed to find program exception', e);
-      return RepoError.SERVER_ERROR;
+      throw new DatabaseError('Cannot save program exception.');
```

Comment:
I think you've meant "cannot find"?
```suggestion
      throw new DatabaseError('Cannot find program exception.');
```